test/e2e/agent/agent_updates_test.go

```diff
@@ -0,0 +1,84 @@
+package agent_test
+
+import (
+	"strings"
+
+	"github.com/flightctl/flightctl/api/v1alpha1"
+	"github.com/flightctl/flightctl/test/harness/e2e"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/sirupsen/logrus"
+)
+
+var _ = Describe("VM Agent behavior during updates", func() {
+	var (
+		harness  *e2e.Harness
+		deviceId string
+	)
+
+	BeforeEach(func() {
+		harness = e2e.NewTestHarness()
+		deviceId = harness.StartVMAndEnroll()
+	})
+
+	AfterEach(func() {
+		harness.Cleanup(true)
+	})
+
+	Context("updates", func() {
+		It("should update to the requested image", Label("updates"), func() {
+
+			// Check the device status right after bootstrap
+			response := harness.GetDeviceWithStatusSystem(deviceId)
+			device := response.JSON200
+			Expect(device.Status.Summary.Status).To(Equal(v1alpha1.DeviceSummaryStatusType("Online")))
+			Expect(*device.Status.Summary.Info).To(Equal("Bootstrap complete"))
+			Expect(device.Status.Updated.Status).To(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
+
+			var newImageReference string
+
+			harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+				currentImage := device.Status.Os.Image
+				logrus.Infof("current image for %s is %s", deviceId, currentImage)
+				parts := strings.Split(currentImage, ":")
+				imageTag := parts[len(parts)-1]
+				Expect(imageTag).To(Equal("base"))
+
+				// Request an update to the v2
+				parts[len(parts)-1] = "v2"
+				newImageReference = strings.Join(parts, ":")
+				device.Spec.Os = &v1alpha1.DeviceOSSpec{Image: newImageReference}
+				logrus.Infof("updating %s to image %s", deviceId, device.Spec.Os.Image)
+			})
+
+			harness.WaitForDeviceCondition(deviceId, "the device is upgrading to renderedVersion: 2",
+				func(device *v1alpha1.Device) bool {
+					for _, condition := range device.Status.Conditions {
+						if condition.Type == "Updating" && condition.Reason == "Update" && condition.Status == "True" {
+							return true
+						}
+					}
+					return false
+				}, "1m")
+
+			harness.WaitForDeviceCondition(deviceId, "the device is rebooting",
+				func(device *v1alpha1.Device) bool {
+					for _, condition := range device.Status.Conditions {
+						if condition.Type == "Updating" && condition.Reason == "Rebooting" && condition.Status == "True" {
+							return true
+						}
+					}
+					return false
+				}, "2m")
+
+			harness.WaitForDeviceCondition(deviceId, "status.Os.Image gets updated",
+				func(device *v1alpha1.Device) bool {
+					return device.Status.Os.Image == newImageReference
+				}, "2m")
+
+			// TODO(hexfusion): we were expecting this update status not to be unknown at this point
+			// Expect(device.Status.Updated.Status).ToNot(Equal(v1alpha1.DeviceUpdatedStatusType("Unknown")))
```

Comment:
ack:  this is not yet implemented will track can you please ref the BZ
https://issues.redhat.com/browse/EDM-679

Reply:
done.